synapse/replication/tcp/protocol.py

```diff
@@ -254,13 +255,26 @@ async def handle_command(self, cmd: Command):
         # specific handling.
         cmd_func = getattr(self, "on_%s" % (cmd.NAME,), None)
         if cmd_func:
-            await cmd_func(cmd)
+            cmd_func(cmd)
             handled = True
 
         # Then call out to the handler.
         cmd_func = getattr(self.command_handler, "on_%s" % (cmd.NAME,), None)
         if cmd_func:
-            await cmd_func(self, cmd)
+            res = cmd_func(self, cmd)
+
+            # the handler might be a coroutine: fire it off as a background process
+            # if so.
+
+            if hasattr(res, "__await__"):
+
+                async def handle_command():
+                    await res
+
+                run_as_background_process(
+                    "replication-" + cmd.get_logcontext_id(), handle_command
```

Comment:
Can we use `lambda: res`, or define a `async def handle_command(res): await res` so that we don't keep creating function instances here?

Reply:
done!